bucket4j/src/main/java/com/linecorp/armeria/server/throttling/tokenbucket/BandwidthLimit.java

```diff
@@ -0,0 +1,210 @@
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.throttling.tokenbucket;
+
+import static java.util.Objects.requireNonNull;
+
+import java.time.Duration;
+
+import com.google.common.base.MoreObjects;
+
+import io.github.bucket4j.Bandwidth;
+import io.github.bucket4j.Refill;
+
+/**
+ * Stores configurations of a single Token-Bucket bandwidth limit.
+ */
+public class BandwidthLimit {
+
+    private final long limit;
+    private final long overdraftLimit;
+    private final long initialSize;
+    private final Duration period;
+
+    BandwidthLimit(long limit, long overdraftLimit, long initialSize, Duration period) {
+        // validate limit
+        if (limit <= 0L) {
+            throw new IllegalArgumentException("Bandwidth limit must be positive. Found: " + limit);
+        }
+        this.limit = limit;
+
+        // validate overdraftLimit
+        if (overdraftLimit > 0L && overdraftLimit <= limit) {
+            throw new IllegalArgumentException("Overdraft limit has to exceed bandwidth limit " + limit +
+                                               ". Found: " + overdraftLimit);
+        }
+        this.overdraftLimit = overdraftLimit;
+
+        // validate initialSize
+        this.initialSize = initialSize;
+
+        requireNonNull(period, "period");
+        if (period.isNegative() || period.isZero()) {
+            throw new IllegalArgumentException("Bandwidth period must be positive. Found: " + period);
+        }
+        this.period = period;
+    }
+
+    /**
+     * Creates new {@link BandwidthLimit}. Specifies limitation in
+     * <a href="https://github.com/vladimir-bukhtoyarov/bucket4j/blob/1.3/doc-pages/token-bucket-brief-overview.md#token-bucket-algorithm">classic
+     * interpretation</a> of token-bucket algorithm.
+     *
+     * @param limit          the bucket size - defines the count of tokens which can be held by bucket
+     *                       and defines the speed at which tokens are regenerated in bucket
+     * @param overdraftLimit defines the maximum overdraft count of tokens which can be held by
+     *                       bucket, this must exceed the {@code limit}
+     * @param initialSize   the initial number of tokens for this bandwidth
+     * @param period         the time window, during which the tokens will be regenerated
+     */
+    public static BandwidthLimit of(long limit, long overdraftLimit, long initialSize, Duration period) {
```

Comment:
Could you move all static factory methods before member field declarations?

Reply:
Fixed